app/api/emojis/index.js

```diff
@@ -0,0 +1,104 @@
+const express = require('express');
+const Emojis = require('../../database/models/emojiModel');
+const Users = require('../../database/models/usersModel');
+
+const logger = require('../../logger');
+
+const checkFields = require('../../../util/checkFields');
+
+const router = express.Router();
+
+// Get Emojis by post id
+router.put('/', async (req, res) => {
+  const requiredFields = ['tableName', 'postId'];
+
+  const error = checkFields(requiredFields, req.body);
+
+  if (error) return res.status(400).json({ message: error });
+
+  try {
+    const emojis = await Emojis.findByPost(req.body.tableName, req.body.postId);
+
+    return res.status(200).json(emojis);
+  } catch (err) {
+    logger.error(JSON.stringify(err));
+    return res.status(500).json({ message: err.message });
+  }
+});
+
+// Get Emoji reactions by post ID, by emoji (Who reacted with this emoji)
+router.put('/:emoji', async (req, res) => {
+  const { emoji } = req.params;
+
+  const requiredFields = ['tableName', 'postId'];
+
+  const error = checkFields(requiredFields, req.body);
+
+  if (error) return res.status(400).json({ message: error });
+
+  try {
+    const userIds = await Emojis.findUserIdsByReaction(
+      req.body.tableName,
+      req.body.postId,
+      emoji,
+    );
+
+    const users = await Users.getNameAndAvatarByIds(
+      userIds.map((usr) => usr.user_id),
+    );
+
+    return res.status(200).json(users);
+  } catch (err) {
+    logger.error(JSON.stringify(err));
+    return res.status(500).json({ message: err.message });
+  }
+});
+
+// Add emoji reaction to post
+router.post('/', async (req, res) => {
+  const requiredFields = ['tableName', 'postId', 'emoji'];
+
+  const error = checkFields(requiredFields, req.body);
+
```

Comment:
**suggestion**: I don't see anywhere where you're actually validating the length of the emoji. The last thing we need is the frontend to crash because someone posted the entire bee movie script as an emoji.

Reply:
Be careful with this though, you can't just do `emoji.length` since `.length` in JS counts Unicode codepoints. 

From a cursory google search, here's an article about emojis in JS and a regex that should be able to solve this: https://thekevinscott.com/emojis-in-javascript/ Just *please* document the regex if you include it :smile: